corehq/motech/tests/test_value_source.py

```diff
@@ -315,6 +321,88 @@ def test_with_location(self):
             as_value_source({"location_field": "dhis_id"})
 
 
+class TestSupercaseValueSourceValidation(SimpleTestCase):
+
+    def test_supercase_value_source(self):
+        value_source = as_value_source({
+            'supercase_value_source': {'case_property': 'foo'},
+        })
+        self.assertIsInstance(value_source, SupercaseValueSource)
+
+    def test_identifier(self):
+        value_source = as_value_source({
+            'supercase_value_source': {'case_property': 'foo'},
+            'identifier': 'bar',
+        })
+        self.assertIsInstance(value_source, SupercaseValueSource)
+
+    def test_referenced_type(self):
+        value_source = as_value_source({
+            'supercase_value_source': {'case_property': 'foo'},
+            'referenced_type': 'bar',
+        })
+        self.assertIsInstance(value_source, SupercaseValueSource)
+
+    def test_relationship(self):
+        value_source = as_value_source({
+            'supercase_value_source': {'case_property': 'foo'},
+            'relationship': 'extension',
+        })
+        self.assertIsInstance(value_source, SupercaseValueSource)
+
+    def test_relationship_invalid(self):
+        with self.assertRaises(TypeError):
+            as_value_source({
+                'supercase_value_source': {'case_property': 'foo'},
+                'relationship': 'invalid',
+            })
+
+    def test_supercase_value_source_empty(self):
+        with self.assertRaises(TypeError):
+            as_value_source({
+                'supercase_value_source': {},
+            })
+
+    def test_supercase_value_source_missing(self):
+        with self.assertRaises(TypeError):
+            as_value_source({
+                'supercase_value_source': {},
+            })
```

Comment:
This test is identical to the one above it. Was that intentional?

Reply:
Oops! Thank you.

It's supposed to be ...
```python
    def test_supercase_value_source_missing(self):
        with self.assertRaises(TypeError):
            as_value_source({})
```
...  but that's a pretty lame test. (Not only is it identical to a test in the suite for SubcaseValueSource, it's also identical to a test in the ValueSource suite.) I should just drop them.

Reply:
553992b